backend/data_adapter.go

```diff
@@ -33,30 +33,58 @@ func (a *dataSDKAdapter) QueryData(ctx context.Context, req *pluginv2.QueryDataR
 			return RequestStatusFromError(innerErr), innerErr
 		}
 
+		if errors.Is(innerErr, context.Canceled) {
+			return RequestStatusCancelled, nil
+		}
+
 		// Set downstream status source in the context if there's at least one response with downstream status source,
 		// and if there's no plugin error
 		var hasPluginError bool
 		var hasDownstreamError bool
+		var hasCancelledError bool
 		for _, r := range resp.Responses {
 			if r.Error == nil {
 				continue
 			}
+
+			if errors.Is(r.Error, context.Canceled) {
```

Comment:
The sae as above, is it enough to just check `context.Canceled`, or should we also check `grpcstatus.Code(err) == grpccodes.Canceled ` ? Asking cause we are doing that in `RequestStatusFromError`.